### PR TITLE
Add log for delete table API

### PR DESCRIPTION
**Pull request by jackjlli**
This PR adds logs for deleteTable API. We've seen this API taking too much time to proceed. Adding logs helps to track which part is the bottle neck.

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 1]
```diff
@@ -1324,47 +1324,80 @@ public IdealState apply(@Nullable IdealState idealState) {
...
+    long removeFromBrokerResourceFinishTime = System.currentTimeMillis();
+    LOGGER.info("Deleting table {}: Removed from broker resource in {}ms", offlineTableName,
+        (removeFromBrokerResourceFinishTime - startTime));
 
     // Drop the table
     if (_helixAdmin.getResourcesInCluster(_helixClusterName).contains(offlineTableName)) {
       _helixAdmin.dropResource(_helixClusterName, offlineTableName);
     }
+    long dropTableFinishTime = System.currentTimeMillis();
+    LOGGER.info("Deleting table {}: Dropped table in {}ms", offlineTableName,
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: s/Dropped table/Removed helix table resource/

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 2]
```diff
@@ -1324,47 +1324,80 @@ public IdealState apply(@Nullable IdealState idealState) {
...
+        (dropTableFinishTime - removeFromBrokerResourceFinishTime));
 
     // Remove all segments for the table
     _segmentDeletionManager.removeSegmentsFromStore(offlineTableName, getSegmentsFor(offlineTableName));
     ZKMetadataProvider.removeResourceSegmentsFromPropertyStore(_propertyStore, offlineTableName);
+    long removeSegmentFinishTime = System.currentTimeMillis();
+    LOGGER.info("Deleting table {}: Removed segments in {}ms", offlineTableName,
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Removed segment metadata

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 3]
```diff
@@ -1324,47 +1324,80 @@ public IdealState apply(@Nullable IdealState idealState) {
...
+        (removeFromBrokerResourceFinishTime - startTime));
 
     // Cache the state and drop the table
     Set<String> instancesForTable = null;
     if (_helixAdmin.getResourcesInCluster(_helixClusterName).contains(realtimeTableName)) {
       instancesForTable = getAllInstancesForTable(realtimeTableName);
       _helixAdmin.dropResource(_helixClusterName, realtimeTableName);
     }
+    long dropTableFinishTime = System.currentTimeMillis();
+    LOGGER.info("Deleting table {}: Dropped table in {}ms", realtimeTableName,
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Removed helix resource for table

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 4]
```diff
@@ -1324,47 +1324,80 @@ public IdealState apply(@Nullable IdealState idealState) {
...
+        (dropTableFinishTime - removeFromBrokerResourceFinishTime));
 
     // Remove all segments for the table
     _segmentDeletionManager.removeSegmentsFromStore(realtimeTableName, getSegmentsFor(realtimeTableName));
     ZKMetadataProvider.removeResourceSegmentsFromPropertyStore(_propertyStore, realtimeTableName);
+    long removeSegmentFinishTime = System.currentTimeMillis();
+    LOGGER.info("Deleting table {}: Removed segments in {}ms", realtimeTableName,
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Removed segment metadata

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java (hunk 2 above)
Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Actually it removes metadata from ZK AND moves segment files to `Delete_Segment` dir as well. That's why I name it to `Removed segments`

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java (hunk 1 above)
Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Done.

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java (hunk 4 above)
Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Done.

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java (hunk 2 above)
Comment (mcvsubbu) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: ah, ok. Then call it as "Removed segment metadata and stored segments" Maybe we need indepdent steps to determine which part is taking time?

Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Suggestion accepted. 👍

**Pull request merged**